rsconf/component/network.py

```diff
@@ -142,14 +142,15 @@ def unchecked_public_ip(self):
         return None
 
     def _add_ports(self, which, ports):
-        assert len(ports[0]) > 2, "invalid ports: {}".format(ports)
+        l = [str(p) for p in ports]
```

Comment:
The temporary is unnecessary. You could leave the assertion the way it was, because it is clearer (testing arguments before touching them).

Reply:
Is this what you mean be55952?

Reply:
The code actually makes no sense. What it was trying to fix was someone passing a correct port at a time when we weren't passing int ports. Here's where it was added: https://github.com/radiasoft/rsconf/commit/5d25bebff7230ebc2f1c305009d8c2135c68e07d

I see now why you had converted it to a str for a test. However, that test was for the name of the port ("domain", "smtp", etc.).  It would fail incorrectly if the port was 80 or 25 (int). Therefore, the assert should be removed or replaced with something a lot more intelligent: int ports need to be > 0 and str ports need to be found in /etc/services. That would be a nice assertion, but it's really not going to work well.

Furthermore, the assertion later doesn't check for collisions between 443 and https. The most correct software would convert all named ports to ints, and look for collisions.

Not worth implementing right now.

Reply:
Make sense. Thanks for the explanation.